### PR TITLE
feat:android native ad support

### PR DESCRIPTION
**Pull request by hexinghuaxd**

### CODE REVIEW

File: docs/tds-tapad.mdx [hunk 1]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
 
 </MultiLang>
 
+## 自渲染信息流广告 {#TapFeedAd}
+为可自定义布局的信息流广告，包含大图、和视频两种基本样式类型，应用下载、跳转到落地页、跳转到浏览器三种交互类型。
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1074]:
```suggestion
为可自定义布局的信息流广告，包含大图和视频两种基本样式类型，应用下载、跳转到落地页和跳转到浏览器三种交互类型。
```

File: docs/tds-tapad.mdx [hunk 2]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+:::tip
+支持的广告尺寸：
+- 大图(宽高比：1.78的图片)
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1078]:
```suggestion
- 大图(宽高比：1.78 的图片)
```

File: docs/tds-tapad.mdx [hunk 3]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+- 视频(宽高比：1.78的视频)
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1079]:
```suggestion
- 视频(宽高比：1.78 的视频)
```

File: docs/tds-tapad.mdx [hunk 4]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+        // 获取广告失败
+    }
+
+    @Override
+    public void onFeedAdLoad(TapFeedAd feedAd) {
+        
+    }
+});
+
+// TapFeedAd接口说明
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1113]:
```suggestion
// TapFeedAd 接口说明
```

File: docs/tds-tapad.mdx [hunk 5]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+    /**
+     * 广告图标
+     *
+     * @return
+     */
+    String getIconUrl();
+
+    /**
+     * 得到Feed广告图片模式
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1138]:
```suggestion
     * 得到 Feed 广告图片模式
```

File: docs/tds-tapad.mdx [hunk 6]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+    /**
+     * 广告封面图片 Image list
+     *
+     * @return
+     */
+    List<ImageInfo> getImageInfoList();
+    
+    /**
+     * 获取TapTap得分
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1152]:
```suggestion
     * 获取 TapTap 得分
```

File: docs/tds-tapad.mdx [hunk 7]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+     *
+     * @return
+     */
+    double getScore();
+
+    /**
+     * 获取广告的view,如视频广告的view,当前版本为自动播放、默认打开声音，用户可以手动关闭声音。
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1159]:
```suggestion
     * 获取广告的 view，如视频广告的 view，当前版本为自动播放且默认打开声音，用户可以手动关闭声音。
```

File: docs/tds-tapad.mdx [hunk 8]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+     * 获取合规信息
+     *
+     * @return
+     */
+    ComplianceInfo getComplianceInfo();
+
+    /**
+     * 注册可点击的View，click/show会在内部完成
+     *
+     * @param container 渲染广告最外层的ViewGroup
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1175]:
```suggestion
     * @param container 渲染广告最外层的 ViewGroup
```

File: docs/tds-tapad.mdx [hunk 9]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+<>
+
+</>
+
+</MultiLang>
+
+### 广告行为监听
+:::info
+AdInteractionListener涉及到广告计费，必须正确调用，convertView必须使用ViewGroup。
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1255]:
```suggestion
AdInteractionListener 涉及到广告计费，必须正确调用，convertView 必须使用 ViewGroup。
```

File: docs/tds-tapad.mdx [hunk 10]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+:::
+
+在加载到信息流广告后，接入方需要注册在信息流广告中可以点击的View，即TapFeedAd.registerViewForInteraction()方法，以实现广告的功能交互及计。包含图文点击区域的注册，附加创意按钮点击区域的注册，隐私协议区域的注册，应用权限的注册。点击附加创意区域会进行应用下载操作。
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1258]:
```suggestion
在加载到信息流广告后，接入方需要注册在信息流广告中可以点击的 View，即 `TapFeedAd.registerViewForInteraction()` 方法，以实现广告的功能交互及计。包含图文点击区域的注册，附加创意按钮点击区域的注册，隐私协议区域的注册，应用权限的注册。点击附加创意区域会进行应用下载操作。
```

File: docs/tds-tapad.mdx [hunk 11]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+注意： 如果需要点击图文区域也能进行下载，请将图文区域的view传入creativeViews。示例代码如下
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1260]:
```suggestion
注意： 如果需要点击图文区域也能进行下载，请将图文区域的 view 传入 creativeViews。示例代码如下
```

File: docs/tds-tapad.mdx [hunk 12]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+    ...
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        ...
+        if (holder instanceof OtherViewHolder) {
+            ...
+        } else if (holder instanceof adViewHolder) {
+            /**
+            * 注册可点击的View，click/show会在内部完成
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1285]:
```suggestion
            * 注册可点击的 View，click/show 会在内部完成
```

File: docs/tds-tapad.mdx [hunk 13]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            * @param container     渲染广告最外层的ViewGroup
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1286]:
```suggestion
            * @param container     渲染广告最外层的 ViewGroup
```

File: docs/tds-tapad.mdx [hunk 14]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            * @param clickViews    可点击的View的列表
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1287]:
```suggestion
            * @param clickViews    可点击的 View 的列表
```

File: docs/tds-tapad.mdx [hunk 15]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            * @param creativeViews 用于下载或者拨打电话的View
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1288]:
```suggestion
            * @param creativeViews 用于下载或者拨打电话的 View
```

File: docs/tds-tapad.mdx [hunk 16]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            */
+            // 可以被点击的view, 也可以把convertView放进来意味整个item可被点击，点击会跳转到落地页
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1290]:
```suggestion
            // 可以被点击的 view，也可以把 convertView 放进来意味整个 item 可被点击，点击会跳转到落地页
```

File: docs/tds-tapad.mdx [hunk 17]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            List<View> clickViewList = new ArrayList<View>();
+            clickViewList.add(convertView);
+            // 创意点击区域的view 点击根据不同的创意进行下载或拨打电话动作
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1293]:
```suggestion
            // 创意点击区域的 view，点击根据不同的创意进行下载或拨打电话动作
```

File: docs/tds-tapad.mdx [hunk 18]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            //如果需要点击图文区域也能进行下载或者拨打电话动作，请将图文区域的view传入creativeViewList
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1294]:
```suggestion
            //如果需要点击图文区域也能进行下载或者拨打电话动作，请将图文区域的 view 传入 creativeViewList
```

File: docs/tds-tapad.mdx [hunk 19]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+            List<View> creativeViewList = new ArrayList<View>();
+            creativeViewList.add(adViewHolder.mCreativeButton);
+            // 注册普通点击区域，创意点击区域。重要! 这个涉及到广告计费及交互，必须正确调用。convertView必须使用ViewGroup。
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1297]:
```suggestion
            // 注册普通点击区域，创意点击区域。重要！这个涉及到广告计费及交互，必须正确调用。convertView 必须使用ViewGroup。
```

File: docs/tds-tapad.mdx [hunk 20]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+</>
+
+<>
+
+</>
+
+### Feed流广告启动/暂停
+:::info
+如果Feed流是在Activity中使用，需要在Activity的onResume 和 onPause 调用相关方法
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1392]:
```suggestion
如果 Feed 流是在 Activity 中使用，需要在 Activity 的 onResume 和 onPause 调用相关方法
```

File: docs/tds-tapad.mdx [hunk 21]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+如果Feed流是在Fragment中使用，除了要考虑onResume和onPause的系统回调，还要注意Fragment的setUserVisibleHint回调
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1394]:
```suggestion
如果 Feed 流是在 Fragment 中使用，除了要考虑 onResume 和 onPause 的系统回调，还要注意 Fragment 的 setUserVisibleHint 回调
```

File: docs/tds-tapad.mdx [hunk 22]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+<>
+
+</>
+
+### Feed流广告销毁
+:::info
+如果是在Activity中创建，请在Activity中调用销毁方法
+
+如果是在Fragment中创建，请在Fragment中调用销毁方法
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1427]:
```suggestion
如果是在 Activity 中创建，请在 Activity 中调用销毁方法

如果是在 Fragment 中创建，请在 Fragment 中调用销毁方法
```

File: docs/tds-tapad.mdx (hunk 1 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: 已修改

File: docs/tds-tapad.mdx (hunk 2 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: 已修改

File: docs/tds-tapad.mdx (hunk 3 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: 已修改

File: docs/tds-tapad.mdx (hunk 4 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: 已修改

File: docs/tds-tapad.mdx (hunk 5 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 6 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 7 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx [hunk 23]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+
+    /**
+     * 获取合规信息
+     *
+     * @return
+     */
+    ComplianceInfo getComplianceInfo();
+
+    /**
+     * 注册可点击的View，click/show会在内部完成
```
[the remark this answers is not in the document]
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx [hunk 24]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+        // 获取广告失败
+    }
+
+    @Override
+    public void onFeedAdLoad(TapFeedAd feedAd) {
+        
+    }
+});
+
+TapFeedAd 接口说明
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1113]:
```suggestion
// TapFeedAd 接口说明
```

File: docs/tds-tapad.mdx (hunk 8 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 9 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 10 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 11 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 12 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 13 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 14 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 16 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 17 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 18 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 19 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 20 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx (hunk 21 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

File: docs/tds-tapad.mdx [hunk 25]
```diff
@@ -1070,6 +1070,386 @@ parentLayout.addView(bannerAd.getBannerView());
...
+</>
+
+</MultiLang>
+
+### 广告行为监听
+:::info
+AdInteractionListener 涉及到广告计费，必须正确调用，convertView 必须使用 ViewGroup。
+:::
+
+在加载到信息流广告后，接入方需要注册在信息流广告中可以点击的 View，即`TapFeedAd.registerViewForInteraction()`方法，以实现广告的功能交互及计。包含图文点击区域的注册，附加创意按钮点击区域的注册，隐私协议区域的注册，应用权限的注册。点击附加创意区域会进行应用下载操作。
```
Comment (SXiaoXu) [inline docs/tds-tapad.mdx:1258]:
```suggestion
在加载到信息流广告后，接入方需要注册在信息流广告中可以点击的 View，即 `TapFeedAd.registerViewForInteraction()` 方法，以实现广告的功能交互及计。包含图文点击区域的注册，附加创意按钮点击区域的注册，隐私协议区域的注册，应用权限的注册。点击附加创意区域会进行应用下载操作。
```

File: docs/tds-tapad.mdx (hunk 22 above)
Comment (hexinghuaxd) [reply docs/tds-tapad.mdx]: fixed

**Review by SXiaoXu — APPROVED** [review]

**Pull request merged**